src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -39,90 +39,270 @@
+                throw new ArchivePathException(
+                        archivedRecordingsPath.toString(), "is not readable");
+            }
+            if (!fs.isDirectory(archivedRecordingsPath)) {
+                throw new ArchivePathException(
+                        archivedRecordingsPath.toString(), "is not a directory");
+            }
+            WebServer webServer = webServerProvider.get();
+            List<String> subdirectories = this.fs.listDirectoryChildren(archivedRecordingsPath);
+            List<ArchivedRecordingInfo> archivedRecordings = new ArrayList<>();
+            for (String subdirectory : subdirectories) {
+                List<String> files =
+                        this.fs.listDirectoryChildren(archivedRecordingsPath.resolve(subdirectory));
+                List<ArchivedRecordingInfo> temp =
+                        files.stream()
+                                .map(
+                                        file -> {
+                                            try {
+                                                return new ArchivedRecordingInfo(
+                                                        subdirectory,
+                                                        webServer.getArchivedDownloadURL(file),
+                                                        file,
+                                                        webServer.getArchivedReportURL(file));
+                                            } catch (SocketException
+                                                    | UnknownHostException
+                                                    | URISyntaxException e) {
+                                                logger.warn(e);
+                                                return null;
+                                            }
+                                        })
+                                .filter(Objects::nonNull)
+                                .collect(Collectors.toList());
+                archivedRecordings.addAll(temp);
+            }
+            future.complete(archivedRecordings);
+        } catch (ArchivePathException | IOException e) {
+            future.completeExceptionally(e);
+        }
+
+        return future;
+    }
+
+    public Future<Path> getRecordingPath(String recordingName) {
+        CompletableFuture<Path> future = new CompletableFuture<>();
+
+        try {
+            List<String> subdirectories = this.fs.listDirectoryChildren(archivedRecordingsPath);
+            Path archivedRecording =
+                    searchSubdirectories(subdirectories, archivedRecordingsPath, recordingName);
+            if (!(archivedRecording != null
+                    && fs.exists(archivedRecording)
+                    && fs.isRegularFile(archivedRecording)
+                    && fs.isReadable(archivedRecording))) {
+                throw new RecordingNotFoundException("archives", recordingName);
```

Comment:
`RecordingNotFoundException` will be thrown here if the path exists but is either not `regularFile` (so is a directory, probably) or is unreadable. Should this be a differently named exception in this case, perhaps? Or at least, should the specific reason for the failure (not a file, or not readable) be included in the exception?

Reply:
Good catch, the `ArchivePathException` class I created would be more appropriate here.